examples/maps/maps.bal

```diff
@@ -1,43 +1,73 @@
 import ballerina/io;
 
 public function main() {
-    // Declares a `map` constrained by the type `string`.
+    // Declare a `map` constrained by the type `string`.
     map<string> m;
 
-    // You can also declare and initialize a `map` with a map literal.
+    string city = "Colombo";
+    string country = "Sri Lanka";
+    string codeLiteral = "code";
+
+    // You can also declare and initialize a `map` with a mapping constructor
+    // expression.
     map<string> addrMap = {
+        // A field can be specified as a key-value pair.
         line1: "No. 20",
         line2: "Palm Grove",
+        // The value in a key-value pair can be any expression, which evaluates
+        // to a value of a type that belongs to the constraint type of the map.
         city: "Colombo 03",
-        country: "Sri Lanka"
+        // A field can also be just a variable reference, which would result
+        // in the variable name being the field name and the variable itself
+        // being the value expression.
+        // This is equivalent to saying `country: country`.
+        country,
+        // The key in a key-value pair can also be a computed key.
+        // A computed key is defined by specifying the key expression within
+        // brackets. The key expression must belong to type `string`.
+        // For a computed key, the key expression is evaluated at runtime and
+        // the resulting value is used as the key.
+        [codeLiteral]: "00300"
     };
     io:println(addrMap);
 
-    // Retrieves a value using member access.
-    string country = <string>addrMap["country"];
-    io:println(country);
+    // Retrieve a value using member access.
```

Comment:
This first sentence sounds like a heading. Shall we rephrase this or remove it altogether?